aten/src/ATen/CPUApplyUtils.h

```diff
@@ -220,7 +220,7 @@ inline std::string _all_equal_numel_error(at::ArrayRef<Tensor> tensors) {
 }
 
 inline bool _apply_preamble(ArrayRef<Tensor> tensors) {
-  checkBackend("CPU_tensor_apply", tensors, Backend::CPU);
+  checkDeviceType("CPU_tensor_apply", tensors, DeviceType::CPU);
```

Comment:
Is this OK?? @li-roy

Reply:
Can we also ensure layout is strided as well?